dart/dynamics/BodyNode.h

```diff
@@ -613,6 +526,63 @@ class BodyNode :
   template <class NodeType, typename ...Args>
   NodeType* createNode(Args&&... args);
 
+  /// Create an ShapeNode attached to this BodyNode. Pass an
+  /// ShapeNode::Properties argument into this function.
+  template <class ShapeNodeProperties>
+  ShapeNode* createShapeNode(const ShapeNodeProperties& properties);
+
+  /// Create an ShapeNode with the specified name
+  ShapeNode* createShapeNode(
+      const ShapePtr& shape, const std::string& name = "ShapeNode");
+
+  /// Create an ShapeNode with the specified name
+  ShapeNode* createShapeNode(const ShapePtr& shape, const char* name);
+
+  /// Create an ShapeNode with the specified name and addons
+  template <class... Addons>
+  ShapeNode* createShapeNode(const ShapePtr& shape,
+                             const std::string& name = "ShapeNode");
+
+  /// Return the number of all the ShapeNodes in this BodyNode
+  size_t getNumShapeNodes() const;
+
+  /// Return the number of ShapeNodes containing given Addon in this BodyNode
+  template <class Addon>
+  size_t getNumShapeNodes() const;
+
+  /// Return the index-th ShapeNode
+  ShapeNode* getShapeNode(size_t index);
+
+  /// Return the index-th (const) ShapeNode
+  const ShapeNode* getShapeNode(size_t index) const;
+
+  /// Return the list of ShapeNodes
+  const std::vector<ShapeNode*> getShapeNodes();
+
+  /// Return the list of (const) ShapeNodes
+  const std::vector<const ShapeNode*> getShapeNodes() const;
+
+  /// Return the list of ShapeNodes containing given Addon
+  template <class Addon>
+  const std::vector<ShapeNode*> getShapeNodesWith();
+
+  /// Return the list of ShapeNodes containing given Addon
+  template <class Addon>
+  const std::vector<const ShapeNode*> getShapeNodesWith() const;
+
+  /// Remove all ShapeNodes from this BodyNode
+  void removeAllShapeNodes();
+
+  /// Remove all ShapeNodes containing given Addon from this BodyNode
+  template <class Addon>
+  void removeAllShapeNodes();
+
+  /// Add a collision Shape into the BodyNode
+  void addCollisionShapeNode(const ShapePtr& shape);
```

Comment:
It looks like this function has been left undefined. I'm not exactly sure how we should approach defining it. Should the ``ShapeNode`` be given default properties? I'm inclined to say that we don't really need this function, since we already have the templated version of ``createShapeNode`` that allows the collision addon to be created simultaneously.